src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/processaudiovideo/ProcessAudioVideoEvidenceAboutToSubmitHandler.java

```diff
@@ -93,6 +94,94 @@ private void processIfIssueDirectionNotice(SscsCaseData caseData) {
+                } else {
+                    sscsDocuments.add(buildAudioVideoSscsDocument(audioVideoEvidence, response));
+                }
+            }
+
+            if (caseData.getDwpDocuments() != null) {
+                dwpDocuments.addAll(caseData.getDwpDocuments());
+            }
+
+            caseData.setDwpDocuments(dwpDocuments);
+
+            if (caseData.getSscsDocument() != null) {
+                sscsDocuments.addAll(caseData.getSscsDocument());
+            }
+
+            caseData.setSscsDocument(sscsDocuments);
+
+            caseData.setAudioVideoEvidence(null);
+        }
+    }
+
+    private DwpDocument buildAudioVideoDwpDocument(AudioVideoEvidence audioVideoEvidence, PreSubmitCallbackResponse<SscsCaseData> response) {
+
+        DocumentLink rip1Doc = null;
+        if (audioVideoEvidence.getValue().getRip1Document() != null) {
+            rip1Doc = buildRip1Doc(audioVideoEvidence.getValue());
+        }
+
+        return DwpDocument.builder().value(
+                DwpDocumentDetails.builder()
+                        .documentLink(audioVideoEvidence.getValue().getDocumentLink())
+                        .documentFileName(audioVideoEvidence.getValue().getFileName())
+                        .documentType(findAudioVideoDocumentType(audioVideoEvidence, response))
+                        .documentDateAdded(audioVideoEvidence.getValue().getDateAdded().toString())
+                        .partyUploaded(audioVideoEvidence.getValue().getPartyUploaded())
+                        .dateApproved(LocalDate.now().toString())
+                        .rip1DocumentLink(rip1Doc)
+                        .build())
+                .build();
+    }
+
+    private DocumentLink buildRip1Doc(AudioVideoEvidenceDetails audioVideoEvidence) {
+        String rip1FileName = "RIP 1 document uploaded on " + audioVideoEvidence.getDateAdded().toString() + ".pdf";
+
+        return DocumentLink.builder()
+                .documentFilename(rip1FileName)
+                .documentUrl(audioVideoEvidence.getRip1Document().getDocumentUrl())
+                .documentBinaryUrl(audioVideoEvidence.getRip1Document().getDocumentBinaryUrl())
+            .build();
+
+    }
+
+    private SscsDocument buildAudioVideoSscsDocument(AudioVideoEvidence audioVideoEvidence, PreSubmitCallbackResponse<SscsCaseData> response) {
+        return SscsDocument.builder().value(
+                SscsDocumentDetails.builder()
+                        .documentLink(audioVideoEvidence.getValue().getDocumentLink())
+                        .documentFileName(audioVideoEvidence.getValue().getFileName())
+                        .documentType(findAudioVideoDocumentType(audioVideoEvidence, response))
+                        .documentDateAdded(audioVideoEvidence.getValue().getDateAdded().toString())
+                        .partyUploaded(audioVideoEvidence.getValue().getPartyUploaded())
```

Comment:
Why are we using enum as it is for some fields (partyUploaded) and string value of enum (documentType) for some other fields?

Reply:
Document type should be an enum but it's used all over the place so a big job. We should do this as part of tech debt